### PR TITLE
Fix invalid index error

### PR DESCRIPTION
**Pull request by sadortun**

### CODE REVIEW

**Comment by alcohol** [comment]
This normally results in a notice. I think it all depends on how your error reporting is setup and whether or not you are converting notices and such into errors/exceptions or not.

**Comment by sadortun** [comment]
Notice or not, you are still accessing an invalid index, which is bad !

**Comment by stof** [comment]
the question is, in which case is this index not set ?

**Comment by stof** [comment]
OK, `app/config/defaults.yml` sets the parameter to an empty array. So a local setup would indeed not have these keys.

**Comment by sadortun** [comment]
@stof, I would say in the default install case, since this parameter is, to my knowledge, not described anywhere in the install docs, or `parameters.yaml.dist` ...

File: src/Packagist/WebBundle/HealthCheck/MetadataDirCheck.php [hunk 1]
```diff
@@ -20,7 +20,7 @@ public function __construct(array $awsMetadata)
 
     public static function isMetadataStoreMounted(array $awsMeta): bool
     {
-        if ($awsMeta['primary'] && $awsMeta['has_instance_store']) {
+        if (isset($awsMeta['primary']) && isset($awsMeta['has_instance_store'])) {
```
Comment (stof) [inline src/Packagist/WebBundle/HealthCheck/MetadataDirCheck.php]: this is wrong. It won't check the boolean values of these params, which will break the prod (a secondary node would get handled as primary, as `isset($awsMeta['primary'])` would return `true` for `['primary' => false]`

Comment (sadortun) [reply src/Packagist/WebBundle/HealthCheck/MetadataDirCheck.php]: You are right !

Comment (stof) [reply src/Packagist/WebBundle/HealthCheck/MetadataDirCheck.php]: what you should do, to be consistent with other places using `$awsMeta` in the codebase is to change that to `!empty($awsMeta) && $awsMeta['primary'] && ...`

**Comment by sadortun** [comment]
Thanks @Seldaek, sorry for the delay !

**Comment by donaldinou** [comment]
Sorry to come back on this issue but I think that the function "isMetadataStoreMounted" should have the `!empty($awsMeta)` verification because it is call on the `dump` method from the `SymlinkDumper` and so used with the associated command.

**Comment by Seldaek** [comment]
@donaldinou done in 8720c3f

**Pull request closed without merging**